packages/cli/tests/unit/agentInstall/installCommand.spec.ts

```diff
@@ -749,27 +751,50 @@ appmap_dir: tmp/appmap
 
   describe('A JavaScript project', () => {
     const jsTestE2E = async (
+      installerName: string,
       nodeVersion: (command: CommandStruct) => Promise<CommandReturn>,
+      getYarnWorkspaces: (command: CommandStruct) => Promise<CommandReturn>,
       installAgent: (command: CommandStruct) => Promise<CommandReturn>,
       initAgent: (command: CommandStruct) => Promise<CommandReturn>,
       validateAgent: (command: CommandStruct) => Promise<CommandReturn>,
       evalResults: (err: Error | undefined, argv: any, output: string) => void,
       checkCurrentConfig: (command: CommandStruct) => Promise<CommandReturn>
     ) => {
       let callIdx = 0;
-      sinon
-        .stub(commandRunner, 'run')
-        .onCall(callIdx++)
-        .callsFake(nodeVersion)
-        .onCall(callIdx++)
-        .callsFake(checkCurrentConfig)
-        .onCall(callIdx++)
-        .callsFake(installAgent)
-        .onCall(callIdx++)
-        .callsFake(initAgent)
-        .onCall(callIdx++)
-        .callsFake(validateAgent)
-        .onCall(callIdx++);
+      if (installerName === 'yarn') {
+        sinon
+          .stub(commandRunner, 'run')
+          .onCall(callIdx++)
+          .callsFake(nodeVersion)
+          .onCall(callIdx++)
+          .callsFake(getYarnWorkspaces)
+          .onCall(callIdx++)
+          // why must installAgent be here and twice for the yarn test to pass?
+          .callsFake(installAgent)
```

Comment:
Needing to do this seems strange to me.

Reply:
Yes that *is* strange. It would be nice to resolve this before merging.

Reply:
Also strange:
- the test [`installAgent`](https://github.com/getappmap/appmap-js/blob/main/packages/cli/tests/unit/agentInstall/installCommand.spec.ts#L894) function and [`javaScriptAgentInstaller.installAgent`](https://github.com/getappmap/appmap-js/blob/main/packages/cli/src/cmds/agentInstaller/javaScriptAgentInstaller.ts#L107) both get called.
- changing the command issued by the test `installAgent` function (to something like `echo hi`) leads to the testcase not failing.

![functions_called_twice](https://user-images.githubusercontent.com/111290954/216448680-e3561f8a-2681-4980-bc91-de9416af534f.png)

Reply:
Maybe it could be made cleaner with jest instead of sinon. Jest mocking is pretty solid, plus you can chain `.mockImplementationOnce` to build a sequence without interspersing it with `onCall`.

Reply:
> Maybe it could be made cleaner with jest instead of sinon. Jest mocking is pretty solid, plus you can chain .mockImplementationOnce to build a sequence without interspersing it with onCall.

I tried using jest with .mockImplementationOnce  instead of sinon. Same result.